autogluon/contrib/enas/enas.py

```diff
@@ -2,7 +2,7 @@
 import collections
 import mxnet as mx
 from mxnet import gluon
-from ...core.space import Categorical, Space, _strip_config_space
+from autogluon_core.core.space import Categorical, Space, _strip_config_space
```

Comment:
+1 for absolute instead of relative imports, the latter never really worked for me.

Are you sure you want to rename autogluon -> autogluon_core?
Would it not rather be useful to put the non-core parts a little elsewhere, or in a submodule?

Just thinking the core should be mainly what AG is about (and maybe it should include most of AG tabular, except esoteric base learners).
If so, it would be good to call it just "autogluon", no?

It maybe does not matter so much, since the user does not see this ...

Reply:
I am not switching away from relative imports, rather I am separating autogluon into autogluon and autogluon_core.

Things in autogluon will relative import things in autogluon, and absolute import things in autogluon_core, and autogluon_core will relative import things in autogluon_core.

I was originally against relative imports when it was first proposed to me, but I am in support of them now. They make refactoring much simpler and cleaner, and reduce the impact of name collisions.

Regarding what core is: Core is just core functionality for HPO / Searchers / etc. Tabular will not exist in it (nor CV or NLP) at least initially.